### PR TITLE
chore(ssr): delete root index.html

### PR DESCRIPTION
**Pull request by LeoMcA**
## Summary

split from https://github.com/mdn/yari/pull/11672

https://mozilla-hub.atlassian.net/browse/MP-1456

### Problem

We want to test deleting the root index.html independently to catch any issues easily/early.

### Solution

Delete `client/build/index.html` after we're done with it, fix some tests.

---

## How did you test this change?

- `yarn && yarn build:client`
- `ls client/build/index.html` <- it's there!
- `yarn build:ssr`
- `ls client/build/index.html` <- it's no longer there!

Will also deploy to stage.

### CODE REVIEW

**Comment by LeoMcA** [comment]
Deployed to staging

File: ssr/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 include.ts
+!*.js
```
Comment (caugner) [inline ssr/.gitignore:2]:
```suggestion
!*.js

```

File: ssr/rm-index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// temporary script to delete root index.html for testing
+// seperate from https://github.com/mdn/yari/pull/11672
+import fs from "node:fs";
+import path from "node:path";
+
+import paths from "../client/config/paths.js";
+
+const indexHtmlFilePath = path.join(paths.appBuild, "index.html");
+fs.rmSync(indexHtmlFilePath);
```
Comment (caugner) [inline ssr/rm-index.js:9]:
```suggestion
fs.rmSync(indexHtmlFilePath);

```

File: ssr/rm-index.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// temporary script to delete root index.html for testing
+// seperate from https://github.com/mdn/yari/pull/11672
```
Comment (caugner) [inline ssr/rm-index.js:2]:
```suggestion
// separate from https://github.com/mdn/yari/pull/11672
```

File: package.json [hunk 4]
```diff
@@ -25,7 +25,7 @@
     "build:docs": "cross-env NODE_ENV=production NODE_OPTIONS='--no-warnings=ExperimentalWarning --loader ts-node/esm' node build/cli.ts -n",
     "build:glean": "cd client && cross-env VIRTUAL_ENV=venv glean translate src/telemetry/metrics.yaml src/telemetry/pings.yaml -f typescript -o src/telemetry/generated",
     "build:prepare": "yarn build:client && yarn build:ssr && yarn tool popularities && yarn tool spas && yarn tool gather-git-history && yarn tool build-robots-txt",
-    "build:ssr": "cross-env NODE_ENV=production NODE_OPTIONS='--no-warnings=ExperimentalWarning --loader ts-node/esm' node ssr/prepare.ts && cd ssr && webpack --mode=production",
+    "build:ssr": "cross-env NODE_ENV=production NODE_OPTIONS='--no-warnings=ExperimentalWarning --loader ts-node/esm' node ssr/prepare.ts && cd ssr && webpack --mode=production && node rm-index.js",
```
Comment (caugner) [inline package.json:28]: Note that this prevents `yarn build:ssr` from being run multiple times, but as I understand https://github.com/mdn/yari/pull/11672 this shouldn't be an issue once the template moves into `ssr/render`.

**Review by caugner — APPROVED** [review]

**Comment by LeoMcA** [comment]
Happy with results on stage, closing.

**Pull request closed without merging**